### PR TITLE
minimax

### PR DESCRIPTION
**Pull request by aditiagarwal34550**

### CODE REVIEW

File: minimax.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    depth is current depth in game tree. 
+    nodeIndex is index of current node in scores[].
+    if move is of maximizer return true else false
+    leaves of game tree is stored in scores[]  
+    height is maximum height of Game tree
+'''
+
+def minimax (Depth, nodeIndex, isMax, scores, height):  
+
+    if (Depth == height):
```
Comment (cclauss) [inline minimax.py]: These () are not needed and not helpful.

File: minimax.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+def minimax (Depth, nodeIndex, isMax, scores, height):  
+
+    if (Depth == height):  
+        return scores[nodeIndex] 
+
+    if (isMax): 
+        return max(minimax(Depth + 1, nodeIndex * 2, False, scores, height), minimax(Depth + 1, nodeIndex * 2 + 1, False, scores, height)) 
+    return min(minimax(Depth + 1, nodeIndex * 2, True, scores, height), minimax(Depth + 1, nodeIndex * 2 + 1, True, scores, height)) 
+
```
Comment (cclauss) [inline minimax.py]: Insert `if __name__ == '__main__':  and then indent all other lines below it.  This allows us to run tests without running your code.

File: minimax.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    height is maximum height of Game tree
+'''
+
+def minimax (Depth, nodeIndex, isMax, scores, height):  
+
+    if (Depth == height):  
+        return scores[nodeIndex] 
+
+    if (isMax): 
+        return max(minimax(Depth + 1, nodeIndex * 2, False, scores, height), minimax(Depth + 1, nodeIndex * 2 + 1, False, scores, height))
```
Comment (cclauss) [inline minimax.py]: These lines are too long to be read on GitHub without scrolling left and right so wrap the whole return value in () and then wrap the lines after the comma to improve readability.

File: minimax.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    leaves of game tree is stored in scores[]  
+    height is maximum height of Game tree
+'''
+
+def minimax (Depth, nodeIndex, isMax, scores, height):  
+
+    if (Depth == height):  
+        return scores[nodeIndex] 
+
+    if (isMax):
```
Comment (cclauss) [inline minimax.py]: These () are not needed and not helpful.

**Comment by aditiagarwal34550** [comment]
@cclauss  Thank You for the code review. 
All changes are done. 
Please review the changes.
Thanks.

**Pull request merged**